lib/core/src/recover/model.rs

```diff
@@ -303,10 +314,15 @@ impl TryFrom<ReceiveSwap> for ReceiveSwapImmutableData {
         ))?;
 
         let swap_id = swap.id;
+        let blinding_key = create_response
+            .blinding_key
+            .and_then(|b| SecretKey::from_str(&b).ok())
+            .ok_or(anyhow!("Missing blinding key"))?;
         Ok(ReceiveSwapImmutableData {
             swap_id,
             swap_timestamp: swap.created_at,
             timeout_block_height: create_response.timeout_block_height,
+            blinding_key,
```

Comment:
Not for the scope of this PR of course, just adding a note here that perhaps we consider to set the swap object itself instead of keep adding fields.